test/unit/spec/EnrollWebauthn_spec.js

```diff
@@ -0,0 +1,304 @@
+  var tick = Expect.tick;
+  var testAttestationObject = 'c29tZS1yYW5kb20tYXR0ZXN0YXRpb24tb2JqZWN0';
+  var testClientData = 'c29tZS1yYW5kb20tY2xpZW50LWRhdGE=';
+
+  Expect.describe('EnrollWebauthn', function () {
+
+    function setup(startRouter, onlyWebauthn) {
+      var settings = {};
+      settings['features.webauthn'] = true;
+
+      var setNextResponse = Util.mockAjax();
+      var baseUrl = 'https://foo.com';
+      var authClient = new OktaAuth({url: baseUrl});
+      var successSpy = jasmine.createSpy('success');
+      var router = new Router(_.extend({
+        el: $sandbox,
+        baseUrl: baseUrl,
+        authClient: authClient,
+        globalSuccessFn: successSpy
+      }, settings));
+      Util.registerRouter(router);
+      Util.mockRouterNavigate(router, startRouter);
+      return tick()
+      .then(function () {
+        setNextResponse(onlyWebauthn ? resWebauthn : resAllFactors);
+        router.refreshAuthState('dummy-token');
+        return Expect.waitForEnrollChoices();
+      })
+      .then(function () {
+        router.enrollWebauthn();
+        return Expect.waitForEnrollWebauthn({
+          router: router,
+          beacon: new Beacon($sandbox),
+          form: new Form($sandbox),
+          ac: authClient,
+          setNextResponse: setNextResponse,
+          successSpy: successSpy
+        });
+      });
+    }
+
+    function mockWebauthn(){
+      navigator.credentials = { create: function () {} };
+    }
+
+    function mockWebauthnSuccessRegistration() {
+      mockWebauthn();
+      spyOn(navigator.credentials, 'create').and.callFake(function () {
+        var deferred = Q.defer();
+        deferred.resolve({
+          response: {
+            attestationObject: CryptoUtil.strToBin(testAttestationObject),
+            clientDataJSON: CryptoUtil.strToBin(testClientData)
+          }
+        });
+        return deferred.promise;
+      });
+    }
+
+    function mockU2fSuccessRegistration() {
+      window.u2f = { register: function () {} };
+
+      spyOn(window.u2f, 'register').and.callFake(function (appId, registerRequests, registeredKeys, callback) {
+        callback({
+          registrationData: testAttestationObject,
+          version: 'U2F_V2',
+          challenge: 'G7bIvwrJJ33WCEp6GGSH',
+          clientData: testClientData
+        });
+      });
+    }
+
+    function mockWebauthnFailureRegistration() {
+      Q.stopUnhandledRejectionTracking();
+      mockWebauthn();
+      spyOn(navigator.credentials, 'create').and.callFake(function () {
+        var deferred = Q.defer();
+        deferred.reject({ message: 'something went wrong' });
+        return deferred.promise;
+      });
+    }
+
+    Expect.describe('Header & Footer', function () {
+      itp('displays the correct factorBeacon', function () {
+        return setup().then(function (test) {
+          expect(test.beacon.isFactorBeacon()).toBe(true);
+          expect(test.beacon.hasClass('mfa-u2f')).toBe(true);
```

Comment:
shouldn't this be `mfa-webauthn`?

Reply:
Webauthn factor is reusing U2F images and texts for beacon.